pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -177,14 +177,14 @@ rec {
            "checkInputs" "installCheckInputs"
            "__impureHostDeps" "__propagatedImpureHostDeps"
            "sandboxProfile" "propagatedSandboxProfile"])
-        // (lib.optionalAttrs (name == "")) {
+        // (lib.optionalAttrs (!(attrs ? name))) {
           name = "${attrs.pname}-${attrs.version}";
         } // (lib.optionalAttrs (stdenv.hostPlatform != stdenv.buildPlatform && !dontAddHostSuffix)) {
```

Comment:
This one also should probably make sure either name is set or pname & version are set.